### ISSUE TITLE
"StoryToolkitAI.X.X.X" is damaged and can't be opened

### ISSUE THREAD

**Issue by cmactavish** [octimot/StoryToolkitAI#11]
I'm getting this following error when trying to open the M1 version (Have tested on M1 MacMini and M1 Max Macbook Pro)
<img width="115" alt="image" src="https://user-images.githubusercontent.com/55484398/198423026-7db00f29-e130-4665-b3d9-e5c6a571a119.png">

Right clicking and selecting 'Open' doesn't work either.

**Comment by octimot** [comment]
Could you try to run it via terminal and copy the errors you're getting?

To do that, open terminal în the folder where you have the app, then run:
`./StoryToolkitAI.0.16.16.M.app/Contents/MacOS/StoryToolkitAI.0.16.16.M1`

**Comment by cmactavish** [comment]
That straight away brings up the same error message with nothing appearing in Terminal. Once I click cancel terminal reads:
`zsh: killed     ./StoryToolkitAI.0.16.16.M1.app/Contents/MacOS/StoryToolkitAI.0.16.16.M1`

**Comment by octimot** [comment]
It looks like, for some reason, your Mac thinks that the app has been tempered with I think, but if you downloaded the zip file from the release page, there's no worry. I'll try to re-compile another standalone soon.

You can try to override that by removing the quarantine meta data. Open Terminal again in the folder with the app and try this before starting the app:
`xattr -d com.apple.quarantine ./StoryToolkitAI.0.16.16.M1.app`

**Comment by octimot** [comment]
You can also try downloading this re-compiled version: https://github.com/octimot/StoryToolkitAI/releases/download/v0.16.16/StoryToolkitAI.0.16.16.3.M1.app.zip

**Comment by cmactavish** [comment]
Recompiled version works! Thanks.

**Comment by octimot** [comment]
Cool! Thanks again for the feedback!

**Comment by octimot** [comment]
This is again weird, thanks for reporting!

I'll try to see what exactly is happening since it looks like there's a problem when I'm freezing the tool into the standalone app...

**Comment by KenzoLuna** [comment]
I have the same problem with the latest version on my Macbook Air M1

**Comment by octimot** [comment]
@KenzoLuna Does `xattr -d com.apple.quarantine ./StoryToolkitAI.0.16.18.M1.app` ran inside the Applications folder via Terminal solve it?

**Comment by KenzoLuna** [comment]
@octimot Yes, thank you :)

**Comment by octimot** [comment]
After digging around, here are some conclusions why `StoryToolkitAI.X.X.X" is damaged and can't be opened...`
happens

## Issue

The latest Mac OS versions require all apps to be signed by the developer. The .app file is being signed while we freeze it into the standalone binary, but some of the underlying packages that we're using do not adhere to the Apple guidelines, so the signature process fails. 

Since this is only related to the standalone releases, non-standalone installations are not affected.

The choice is simple at this point: we either remove those packages and kill some of the app functionality, re-organize them by adhering to the Apple guidelines (but that's messy and involves a lot of work, or leave it like it is for now and hope that the users are able to take the app out of the quarantine using the below command:

## Partial solution

1. Open the Terminal window in the folder where the .app file is stored (usually `Downloads` or `Applications`, depending where it was saved/installed)
2. Run this command:
```
xattr -d com.apple.quarantine ./StoryToolkitAI.0.16.18.M1.app
```
3. Cross fingers and hope that the file is out of the quarantine and now the app can now be run.

### Potential vulnerabilities
If the standalone release was not downloaded from the official StoryToolkitAI page, attackers might alter the app to gain access to the users machine.

If you are not comfortable with this, you can avoid it by using the non-standalone version of the tool and installing it by following [the Mac OS Installation steps outlined here](https://github.com/octimot/StoryToolkitAI/blob/main/INSTALLATION.md#mac-os).


### If anyone has a different solution for this, let us know.

**Comment by rdacode** [comment]
Just letting you know that this is still occurring with the latest M1 (17.5), but that the terminal command fixed it for me.

**Comment by octimot** [comment]
@rdacode 

Thanks for the feedback and please do report this whenever you can on new releases!

We're trying to find a solution, but also keep adding new modules to the bundle which messes up things...

**Comment by octimot** [comment]
## Starting with 0.17.16:
The latest standalone version uses an Apple certificate (without notarization), so **right clicking** and clicking **open** in the drop-down on the second try should solve this without the need to run the above command.

If you still need to run the command to open the tool, please let me know.

**Comment by jcplus** [comment]
I downloaded StoryToolkitAI.0.19.5.arm64 and my os prompt it's damaged. I then run `xattr -d com.apple.quarantine ./StoryToolkitAI.0.19.5.arm64` but prompted: No such xattr: com.apple.quarantine.
So I downlaoded the StoryToolkitAI.0.16.16.3.M1.app from this thread and run, still reveive the 'damaged' error. So I run `xattr -d com.apple.quarantine ./StoryToolkitAI.0.16.16.3.M1.app` and no prompts this time. But running the app still reveive the 'damaged' error.
My OS is Ventura 13.5

**Comment by octimot** [comment]
@jcplus 

Sorry for the stupid question, but are you running the command `xattr -d com.apple.quarantine ./StoryToolkitAI.0.19.5.arm64.app`? in the folder where you downloaded / installed the app? In other words, if you perform a `ls`, do you see the file `StoryToolkitAI.0.19.5.arm64.app` listed in Terminal?

Also, make sure you have `.app` at the end of the file. In your report above you were missing it for the first command.

Cheers!

**Comment by jcplus** [comment]
Yes, the files are in my ~/Downloads folder and I run the commands in the same folder where I can see them when `ls -l`.

**Comment by clpStephen** [comment]
I get an error message that says it can't be opened. It doesn't say damaged. This is on Mac Ventura. I have tried the command you give but it reprots there is no such attribute com.apple.quarantine. I am attempting to evaluate your tool, I would really love for it to work for my organization.

**Comment by octimot** [comment]
> I get an error message that says it can't be opened. It doesn't say damaged. This is on Mac Ventura. I have tried the command you give but it reprots there is no such attribute com.apple.quarantine. I am attempting to evaluate your tool, I would really love for it to work for my organization.

Does this happen even after you right-click and open, according to the Unsafe warnings section under the Installation chapter [here](https://github.com/octimot/StoryToolkitAI/releases/tag/v0.23.2)?

**Comment by octimot** [comment]
@jcplus 

> Yes, the files are in my ~/Downloads folder and I run the commands in the same folder where I can see them when `ls -l`. I'm so positive because I normal use `tab` key to finish the file names especially when they are long. I wouldn't be able to autocomplete the file name if I wasn't in the same folder of the files. :)

True, but you might get the name of the .zip file autocompleted using the `tab` key too!

I have to make sure you're entering the right command since the error you're getting is technically impossible on the `.app` file. The only other reason why this might happen is if the download was actually corrupted or the zip extraction failed somehow.

So, please manually enter the name of the file in the command and check the response. Feel free to copy and paste your terminal command and the exact response here as it is and I'll look into it!
